src/modules/udw/udw.module.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, Fragment, useEffect, createContext } from 'react';
+import PropTypes from 'prop-types';
+import { classnames } from '../common/classnames/classnames';
+import Popup from '../common/popup/popup.component';
+import { loadContentTypes } from './services/universal.discovery.service';
+import { restInfo } from '../common/rest-info/rest.info';
+
+export const ContentTypesContext = createContext();
+const getContentTypesFromResponse = (response) => {
+    if (!response || !response.ContentTypeInfoList) {
+        return {};
+    }
+
+    const contentTypesMap = response.ContentTypeInfoList.ContentType.reduce((total, item) => {
+        total[item._href] = item;
+
+        return total;
+    }, {});
+
+    return contentTypesMap;
+};
+
+const UDWModule = ({ title, onClose, tabs, maxHeight }) => {
+    const [contentTypesMap, setContentTypesMap] = useState(null);
+    // useReducer maybe?
+    const [state, setState] = useState(() => {
```

Comment:
Is it intentional to have here `state` instead of `activeTabId` like in the other places?

Reply:
Good catch!